designs/2019-rule-tester-processors/README.md

```diff
@@ -0,0 +1,94 @@
+-   Start Date: 2019-07-16
+-   RFC PR: (leave this empty, to be filled in later)
+-   Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea)), Brandon Mills ([@btmills](https://github.com/btmills))
+
+# RuleTester supports processor
+
+## Summary
+
+This RFC makes `RuleTester` class supporting `processor` option.
+
+## Motivation
+
+Currently, we cannot test rules with processors. This is inconvenient for plugin rules which depend on processors. For example, [vue/comment-directive](https://github.com/vuejs/eslint-plugin-vue/blob/6751ff47b4ecd722bc2e2436ce6b34a510f92b07/tests/lib/rules/comment-directive.js) rule could not use `RuleTester` to test. Rules that distinguish between physical and virtual filenames [cannot be tested without processors](https://github.com/eslint/eslint/issues/14800).
+
+## Detailed Design
+
+To add `processor` option and `processorOptions` (see #29) to test cases.
+
+```js
+const { RuleTester } = require("eslint")
+const rule = require("../../../lib/rules/example-rule")
+const exampleProcessor = require("../../../lib/processors/example-processor")
+
+const tester = new RuleTester()
+
+tester.run("example-rule", rule, {
+    valid: [
+        {
+            code: `
+                <script>
+                    console.log("Hello")
+                </script>
+            `,
+            processor: exampleProcessor,
+            processorOptions: {},
+        },
+    ],
+    invalid: [],
+})
+```
+
+### § `processor` option
+
+This is a definition object of a processor that the "[Processors in Plugins](https://eslint.org/docs/developer-guide/working-with-plugins#processors-in-plugins)" section describes.
+
+If a processor is given, the tester passes its `preprocess`, `postprocess`, and optional `supportsAutofix` properties as part of `Linter#verify()`'s `filenameOrOptions` options object.
```

Comment:
`!processor.supportsAutofix` value will be passed as `filenameOrOptions.disableFixes`?

Reply:
@frankethan109@gmail.com